### DISCUSSION TITLE
PCB/Hat Template for 5A-75E

### DISCUSSION THREAD

**Discussion by GTCLive** [Peter-van-Tol/LiteX-CNC#60] [Show and tell]
I went for the 5A-75E (Ver8.0) for LiteX-CNC and sharing the DXF (outline) of the board and its 16x Jconnectors, 4P JTAG and the 5V main. I understand those pin-to-pin match aren't easy to CAD and so I thought I'd save someone else the trouble. You should be able to import the DXF into your favorite EDA and customize your own board outline from that. I'm also adding a STEP file of the same in case DXF ain't an option. 

In my case I went for a board-to-board 100P connector and a second hat. Reason for that is because I have 3 (three) 5A-75E boards. One for a 2ax lathe, another for a 4ax mill, and one spare in case of a failure. That way it makes it easy to swap the top hat on whatever 5A-75E board I need. I've used a mating pair connector from Panasonic (P/N: AXK5S00047YG and AXK6S00447YG). 

![5A-75B_v8 0](https://github.com/Peter-van-Tol/LiteX-CNC/assets/91785220/650da518-dffa-4866-a371-b6116a13c43a)

If anyone needs other material related to this board just update this thread and I'll see what I can do. 

[5A-75E_v8.0.DXF.and.STP.zip](https://github.com/Peter-van-Tol/LiteX-CNC/files/13533902/5A-75E_v8.0.DXF.and.STP.zip)

**Comment by Peter-van-Tol** [comment]
Thanks for sharing!

I think this board is meant for a 5A-75E, because the B-revision only has 8 connectors. For the B-revision there is also a HAT available, see [this repo](https://github.com/Peter-van-Tol/Litexcnc-HUB75HAT)

I can help you to create a custom pin-out for your board. Basically this translates the pinout of the HUB75 connectors to the pinout of the Panasonic AXK5S00047Y. This ensure that users don't have to know how the board is routed and also makes sure theat different version are behaving the same. I've done something similar for the HUB75HAT.

**Reply by GTCLive** [reply]
Oh snap, well spotted (tired). Fixed the typos. And man you're so kind, thanks for offering. I'm sorted already pcb-wise however be ready I'm sure many others will take on your offer haha

**Reply by Peter-van-Tol** [reply]
Which software did you use for your PCB? Might be interesting to include this in the HUB75HAT project.

**Reply by GTCLive** [reply]
Problem with this board's *.sch & *.pcb is it's hard wired proprietary i.e. 2pwm, 6encoders, 6stepgen, all gpios yada yada. Firmware fits-all lathe + mill I have here type thing. Another issue is I can't recommend the existing 0.5mm pitch 100P connector (50 insertions rated max), but would advise on a 0.8mm minimum or 1.0mm pitch better rather. I honestly think there are much better ways to DIY, the DXF/template is a good starting point. I can at least confirm the attached outline (DXF) is a 99.95% exact fit. Let me have your ideas if you think anything else could be helpful. 

![schema](https://github.com/Peter-van-Tol/LiteX-CNC/assets/91785220/9dc0f8ca-2ba8-4a57-b956-ab6ca2e56ee6)

Top hat for the lathe (panasonic male conn) will look something like that. That's just a dummy benchtop board I had made to test/probe with, not the one that'll go in the controller box. I plan on a similar arrangement hat setup for the milling machine. As you can see, it's all custom and hard to share as a 'community option/solution'. 

![dummyhat](https://github.com/Peter-van-Tol/LiteX-CNC/assets/91785220/0fd62f4d-ce3e-467e-ad97-c9b329c0ee87)

**Comment by GTCLive** [comment]
Well it's official!! Undoing the breadboard and test board this evening and sending this out for production. **Final** - everything works, clean signals, steady timings, HAL/INI done. Don't know what else to say; THANK YOU!!!

![LitexCNC_2ax](https://github.com/Peter-van-Tol/LiteX-CNC/assets/91785220/87aeaeb8-0540-47f2-aea6-cbb8ac8eca58)

Next: 4ax milling machine.

**Reply by Peter-van-Tol** [reply]
I like the design. Please share the schematics for others to learn from. Personally I'm very interested in the ICs used in this design.

**Reply by GTCLive** [reply]
See attached. 

[5a-75e_lathe.pdf](https://github.com/Peter-van-Tol/LiteX-CNC/files/13630294/5a-75e_lathe.pdf)

**Comment by GTCLive** [comment]
Quick update:

Received some upgraded linear scales (KA500 - 1um) and an incremental rotary encoder with 2500ppr for xmas!! This resolution inevitably forces me to upgrade the optocouplers for high-speed ones. I've used a "LTV-M601" part for the job and updated the above schematic accordingly. I've also bumped the caps on the outputs up from a 0805 part to 1206 (TAJA106K016RNJ). 

Note1: if you plan on using affordable sensing hardware (e.g. 100ppr rotary encoder and some KA300 5um like I previously had), the first iteration of the schematic with the slow speed EL3H7 is perfectly fine. Although recommend the second iteration with the "LTV-M601" part, in case you plan on an upgrade later. 

Note2: The MPG interface/inputs should also be isolated (**ideally**). It is not on this schematic. As long as you keep it far from the high voltage noise (motors/spindle/ect) it should be fine as-is. 

**See attached:** 
[5a-75e_lathe-high-speed.pdf](https://github.com/Peter-van-Tol/LiteX-CNC/files/13777663/5a-75e_lathe-high-speed.pdf)

**Reply by EhsanRasouli1371** [reply]
can you send me the editable Altium file that only shows the position of the 5A-75E connectors on it?

**Comment by EhsanRasouli1371** [comment]
The header board schematic I designed for the 5a-75e is as follows.
The problem I have in using it is that when the Linux CNC software is closed, it moves the motors by itself.
Can anyone tell me what the problem is?
![۳۳۳](https://github.com/Peter-van-Tol/LiteX-CNC/assets/147668561/4009c4d8-9076-4e55-bc33-bb3a6d4836b0)
![۳۳۲](https://github.com/Peter-van-Tol/LiteX-CNC/assets/147668561/762cd8e3-d890-48a5-b61c-754e79b4f9f3)

[Schematic Prints.pdf](https://github.com/Peter-van-Tol/LiteX-CNC/files/15257581/Schematic.Prints.pdf)

[123-2024-05-08_05.41.43.zip](https://github.com/Peter-van-Tol/LiteX-CNC/files/15257642/123-2024-05-08_05.41.43.zip)
